File: jung-visualization/src/test/java/edu/uci/ics/jung/visualization/control/TestCrossoverScalingControl.java
package edu.uci.ics.jung.visualization.control;

import com.google.common.graph.Network;
import edu.uci.ics.jung.graph.util.TestGraphs;
import edu.uci.ics.jung.layout.algorithms.FRLayoutAlgorithm;
import edu.uci.ics.jung.visualization.BasicVisualizationServer;
import edu.uci.ics.jung.visualization.VisualizationServer;
import java.awt.*;
import java.awt.geom.Point2D;
import junit.framework.TestCase;

public class TestCrossoverScalingControl extends TestCase {

  CrossoverScalingControl sc;
  VisualizationServer<?, ?> vv;

  float crossover;
  float scale;

  @SuppressWarnings({"rawtypes", "unchecked"})
  @Override
  public void setUp() {
    sc = new CrossoverScalingControl();
    Network<?, ?> network = TestGraphs.getDemoGraph();
    vv = new BasicVisualizationServer(network, new FRLayoutAlgorithm(), new Dimension(600, 600));
  }

  public void testCrossover() {
    crossover = 2.0f;
    scale = .5f;
    sc.setCrossover(crossover);
    sc.scale(vv, scale, new Point2D.Double());
    //		System.err.println("crossover="+crossover);
    //		System.err.println("scale="+scale);
    //		System.err.println("layout scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.LAYOUT).getScale());
    //		System.err.println("view scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.VIEW).getScale());
  }

  public void testCrossover2() {
    crossover = 2.0f;
    scale = 1.5f;
    sc.setCrossover(crossover);
    sc.scale(vv, scale, new Point2D.Double());
    //		System.err.println("crossover="+crossover);
    //		System.err.println("scale="+scale);
    //		System.err.println("layout scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.LAYOUT).getScale());
    //		System.err.println("view scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.VIEW).getScale());

  }

  public void testCrossover3() {
    crossover = 2.0f;
    scale = 2.5f;
    sc.setCrossover(crossover);
    sc.scale(vv, scale, new Point2D.Double());
    //		System.err.println("crossover="+crossover);
    //		System.err.println("scale="+scale);
    //		System.err.println("layout scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.LAYOUT).getScale());
    //		System.err.println("view scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.VIEW).getScale());
  }

  public void testCrossover4() {
    crossover = 0.5f;
    scale = 2.5f;
    sc.setCrossover(crossover);
    sc.scale(vv, scale, new Point2D.Double());
    //		System.err.println("crossover="+crossover);
    //		System.err.println("scale="+scale);
    //		System.err.println("layout scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.LAYOUT).getScale());
    //		System.err.println("view scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.VIEW).getScale());
  }

  public void testCrossover5() {
    crossover = 0.5f;
    scale = .3f;
    sc.setCrossover(crossover);
    sc.scale(vv, scale, new Point2D.Double());
    //		System.err.println("crossover="+crossover);
    //		System.err.println("scale="+scale);
    //		System.err.println("layout scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.LAYOUT).getScale());
    //		System.err.println("view scale =
    // "+vv.getRenderContext().getMultiLayerTransformer().getTransformer(Layer.VIEW).getScale());
  }
}

File: jung-visualization/src/test/kotlin/edu/uci/ics/jung/visualization/control/TestCrossoverScalingControl.kt
package edu.uci.ics.jung.visualization.control

import edu.uci.ics.jung.graph.util.TestGraphs
import edu.uci.ics.jung.layout.algorithms.FRLayoutAlgorithm
import edu.uci.ics.jung.visualization.BasicVisualizationServer
import edu.uci.ics.jung.visualization.VisualizationServer
import java.awt.Dimension
import java.awt.geom.Point2D
import junit.framework.TestCase

class TestCrossoverScalingControl : TestCase() {

  lateinit var sc: CrossoverScalingControl
  lateinit var vv: VisualizationServer<*, *>

  var crossover = 0f
  var scale = 0f

  override fun setUp() {
    sc = CrossoverScalingControl()
    val network = TestGraphs.getDemoGraph()
    vv = BasicVisualizationServer(network, FRLayoutAlgorithm(), Dimension(600, 600))
  }

  fun testCrossover() {
    crossover = 2.0f
    scale = .5f
    sc.crossover = crossover.toDouble()
    sc.scale(vv, scale, Point2D.Double())
  }

  fun testCrossover2() {
    crossover = 2.0f
    scale = 1.5f
    sc.crossover = crossover.toDouble()
    sc.scale(vv, scale, Point2D.Double())
  }

  fun testCrossover3() {
    crossover = 2.0f
    scale = 2.5f
    sc.crossover = crossover.toDouble()
    sc.scale(vv, scale, Point2D.Double())
  }

  fun testCrossover4() {
    crossover = 0.5f
    scale = 2.5f
    sc.crossover = crossover.toDouble()
    sc.scale(vv, scale, Point2D.Double())
  }

  fun testCrossover5() {
    crossover = 0.5f
    scale = .3f
    sc.crossover = crossover.toDouble()
    sc.scale(vv, scale, Point2D.Double())
  }
}
